### ISSUE TITLE
Fix Neo4j Custom Queries getting service deployment information with services

### ISSUE THREAD

**Issue by davidkopp** [UST-MICO/mico#578]
The Neo4j custom queries of `ServiceDeploymentInfoRepository` does not return the included *MicoService* (property `service` is always null). Nevertheless `applicationRepository.findByShortNameAndVersion` works like expected.

`serviceDeploymentInfoRepository.findAllByApplication`:
```
[MicoServiceDeploymentInfo(id=40, service=null, replicas=1, labels=[MicoLabel(id=61, key=component, value=backend)], environmentVariables=[], imagePullPolicy=IfNotPresent, restartPolicy=OnFailure, kubernetesDeploymentInfo=null), MicoServiceDeploymentInfo(id=3, service=null, replicas=1, labels=[MicoLabel(id=44, key=component, value=frontend)], environmentVariables=[MicoEnvironmentVariable(id=43, name=BACKEND_REST_API, value=conduit.productionready.io/api)], imagePullPolicy=IfNotPresent, restartPolicy=OnFailure, kubernetesDeploymentInfo=null)]
```

`serviceDeploymentInfoRepository.findByApplicationAndService`:
```
MicoServiceDeploymentInfo(id=40, service=null, replicas=1, labels=[MicoLabel(id=61, key=component, value=backend)], environmentVariables=[], imagePullPolicy=IfNotPresent, restartPolicy=OnFailure, kubernetesDeploymentInfo=null)
```

`serviceDeploymentInfoRepository.findByApplicationAndServiceAndVersion`:
```
MicoServiceDeploymentInfo(id=40, service=null, replicas=1, labels=[MicoLabel(id=61, key=component, value=backend)], environmentVariables=[], imagePullPolicy=IfNotPresent, restartPolicy=OnFailure, kubernetesDeploymentInfo=null)
```

**Comment by buehlefs** [comment]
We had a similar problem with service interfaces. The neo4j ogm mapper cannot use custom querys like the normal ones. Depth for example will NOT work. So the ogm can only really get one object layer with a custom query. Nested objects will always be null. Also having relationships in the custom query result does not work in any usable way.
The only workaround i found was to use a object a normal query works for and extract the wanted object from this object. Then it will be fully populated from the db.

**Comment by davidkopp** [comment]
Jakob already fixed this with commit [ca2a03eef826ac88e9b6a76a97c84613d8405061](https://github.com/UST-MICO/mico/commit/ca2a03eef826ac88e9b6a76a97c84613d8405061)

**Comment by JakobBenz** [comment]
I already solved the problem. You can return paths from custom queries, then you will get fully hydrated domain objects. For an example see the queries in the service deployment repository (on branch 438).

I also had a custom query working for service interfaces, ports and everything included.  However I don’t know if it’s already in the master.
